Supertext.Base.Security.Cryptography/Hashing/Sha256Hasher.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Text;
+using Supertext.Base.Security.Cryptography.Hashing.Salt;
+using Supertext.Base.Security.Hashing;
+
+namespace Supertext.Base.Security.Cryptography.Hashing
+{
+    internal class Sha256Hasher : ISha256Hasher
+    {
+        private readonly ISaltGenerator _saltGenerator;
+        private readonly HashingConfig _hashingConfig;
+
+        public Sha256Hasher(HashingConfig hashingConfig, ISaltGenerator saltGenerator)
+        {
+            _saltGenerator = saltGenerator;
+            _hashingConfig = hashingConfig;
+        }
+
+        public HashingResult HashLegacyToken(string entry)
+        {
+            return ComputeSha256HashWithSaltAndPepper(entry, _saltGenerator.Generate(), _hashingConfig.LegacyTokenHashingPepper);
+        }
+
+        public HashingResult HashPassword(string entry)
+        {
+            return ComputeSha256HashWithSaltAndPepper(entry, _saltGenerator.Generate(), _hashingConfig.PasswordHashingPepper);
+        }
+
+        public HashingResult ComputeSha256HashWithSaltAndPepper(string rawData, string salt, string pepper)
+        {
+            var encoding = new UnicodeEncoding();
+            var spicedRawData = salt + rawData + pepper;
+            var bytes = encoding.GetBytes(spicedRawData);
+            var sha256 = new SHA256Managed();
```

Comment:
I read that is more widely used and also the 512 version doesn't provide a significant amount of extra security. But I can change it since we are currently using the 512 to hash the passwords in legacy. What do you thing?